### DISCUSSION TITLE
ListBox Drag and Drop: How to Add Visual Feedback?

### DISCUSSION THREAD

**Discussion by artwe** [AvaloniaUI/Avalonia#17730] [Q&A]
I know that the Avalonia.Xaml.Behaviors package provides a Drag and Drop implementation for ListBox Control. I would like to know how to add some visual feedback, such as the **drag item following the mouse**, and **the target item touched by the mouse can be highlighted.**

**Comment by timunie** [comment]
Download the source of that project and explore the samples and tests. It really has some nice impl available

**Comment by metaone01** [comment]
Just a guess:
What if setting the `ItemsSource` like
```
{"cat",
 "camel", 
 "cow",
 "chameleon",
 "",    <- Add an empty item under the dragging control
 "mouse",
 "lion",
 "zebra"}
```
> When dragging, just change the empty item's index, and then an empty `ListBoxItem` shows as visual feedback. 

> When drag ends, just replace the empty item with the one that has been dragged.

Get the correct index is difficult though.
